src/utils.jl

```diff
@@ -77,6 +77,55 @@ function sideof(p::Point{2,T}, c::Chain{2,T}) where {T}
   ifelse(isapprox(w, zero(T), atol=atol(T)), :OUTSIDE, :INSIDE)
 end
 
+"""
+    sideof(point, mesh)
+
+Determines whether a `point` is inside, outside or on the surface of a `mesh`.
+Possible results are `:INSIDE`, `:OUTSIDE`, or `:ON`.
+
+### Notes
+
+Uses a ray-casting algorithm.
+"""
+function sideof(point::Point{3,T}, mesh::Mesh{3,T}) where {T}
+  if !(eltype(mesh) <: Triangle)
+    throw(ArgumentError(
+          "This function only works for surface meshes with triangles as elements."))
+  end
+
+  z = last.(coordinates.(extrema(mesh)))
+  r = Ray(point, Vec(zero(T), zero(T), 2 * (z[2] - z[1])))
+
+  intersects = false
+  edgecrosses = 0
+  cornerscrossing = Point{3,T}[]
+  for e in mesh
+    result = intersection(r, e) do I
```

Comment:
I don't think this do block returns something meaningful? Should we just erase the `result` variable? The logic of this block is still a bit confusing. It does a bunch of checks and then interrupts the loop with explicit `return` statements, but why do we need to save the result if nothing is actually stored as an intersection?

Reply:
Perhaps you can help me here - its the only way I knew to make it work. The check if the point is `:ON` the surface (lines 106-109) should just return the final results immediately, so I don't know how to break out of the do block other than returning the results in the do block and then checking if the result from the do block is `:ON` or `nothing` and if `!isnothing` then return `:ON` (line 121).